src/s1reader/s1_burst_slc.py

```diff
@@ -579,6 +585,332 @@ def az_carrier_components(self, offset, position):
+        tau0_ka_interp = interpolator_tau0_ka(vec_t)[..., np.newaxis]
+
+        interpolator_tau0_fdc_interp = InterpolatedUnivariateSpline(
+                                        vec_aztime_dc_sec,
+                                        self.extended_coeffs.vec_tau0_dc,
+                                        k=1)
+        tau0_fdc_interp = interpolator_tau0_fdc_interp(vec_t)[..., np.newaxis]
+
+        # add range time origin to vec_tau
+        grid_tau += tau0_ka_interp * np.ones(vec_tau.shape)[np.newaxis, ...]
+
+
+        # Interpolate the DC and fm rate coeffs along azimuth time
+        list_a_interp = [] # Doppler centroid
+        list_b_interp = [] # Fm rate
+        for i in range(3):
+            interpolator_coeff_a = InterpolatedUnivariateSpline(
+                                        vec_aztime_dc_sec,
+                                        self.extended_coeffs.lut_coeff_dc[:, i],
+                                        k=1)
+            list_a_interp.append(interpolator_coeff_a(vec_t))
+
+            interpolator_coeff_b = InterpolatedUnivariateSpline(
+                                        vec_aztime_fm_rate_sec,
+                                        self.extended_coeffs.lut_coeff_fm_rate[:, i],
+                                        k=1)
+            list_b_interp.append(interpolator_coeff_b(vec_t))
+
+        arr_coeff_a = np.array(list_a_interp).transpose()
+        arr_coeff_b = np.array(list_b_interp).transpose()
+
+        # Run topo on scratch directory
+        dem_raster = isce3.io.Raster(path_dem)
+        epsg = dem_raster.get_epsg()
+        proj = isce3.core.make_projection(epsg)
+        ellipsoid = proj.ellipsoid
+        grid_doppler = isce3.core.LUT2d()
+
+        Rdr2Geo = isce3.geometry.Rdr2Geo
+
+        # TODO Add parameters (# interation, threshold, etc.) to rdr2geo
+        rdr2geo_obj = Rdr2Geo(
+                radargrid_correction,
+                self.orbit,
+                ellipsoid,
+                grid_doppler,
+                threshold=threshold_rdr2geo,
+                numiter=numiter_rdr2geo)
+
+        str_datetime = datetime.datetime.now().strftime('%Y%m%d_%H%M%S')
+
+        list_filename_llh = [f'{path_scratch}/lat_{str_datetime}.rdr',
+                        f'{path_scratch}/lon_{str_datetime}.rdr',
+                        f'{path_scratch}/hgt_{str_datetime}.rdr']
+        
+        lat_raster, lon_raster, hgt_raster = [isce3.io.Raster(
+                                                        filename_raster,
+                                                        radargrid_correction.width,
+                                                        radargrid_correction.length,
+                                                        1,
+                                                        gdal.GDT_Float64,
+                                                        'ENVI')
+                                              for filename_raster in list_filename_llh]
```

Comment:
```suggestion
        lat_raster, lon_raster, hgt_raster = [isce3.io.Raster(
                                                        f'{path_scratch}/{llh}_{str_datetime}.rdr'],
                                                        radargrid_correction.width,
                                                        radargrid_correction.length,
                                                        1,
                                                        gdal.GDT_Float64,
                                                        'ENVI')
                                              for llh in ['lat', 'lon', 'hgt']]
```
nit - fold in file name generation